samples/net/lwm2m_client/Kconfig

```diff
@@ -27,10 +27,14 @@ config LWM2M_APP_TLS_TAG
 	  TLS security tag to use when connecting.
 endif
 
+config HAS_INTERNET
```

Comment:
This sounds a bit optimistic :smile: should we call this `HAS_CONNECTIVITY` instead.

It is unfortunate that this sample did not use the `NET_SAMPLE` prefix for Kconfig symbols like what is done in many other networking samples. Now `LWM2M_` prefix is used  which gives the impression that the option is part of LWM2M library which this sample is not.

Reply:
Well... those emulated platforms have connectivity too.
I tried to somehow separate the defaults that might actually have global internet connectivity and use public Leshan address.


I see, the `LWM2M_APP` is probably not the best.. maybe I should refactor to `NET_SAMPLE_LWM2M` instead.

Reply:
`config NET_SAMPLE_MIGHT_HAVE_INTERNET`